week2Task1

lapitan opened the pull request:
first commit of week2Task1

lapitan committed:
week2Task1

first commit of week2Task1

Lada303 commented on ulabEdu/src/homework/ComplexExamples.java:
```diff
@@ -0,0 +1,199 @@
+                .entrySet()
+                .stream()
+                .peek(stringLongEntry -> System.out.println("Key: "+stringLongEntry.getKey()+"\nValue: "+stringLongEntry.getValue()))
+                .collect(Collectors.toMap(Map.Entry::getKey,Map.Entry::getValue));
```
вместо peek - forEach, тогда не нежно последнее collеct делать, оно лишнее. Нет проверки на нулл, и нет сортировки по id

lapitan committed:
week2Task1: Fix

added PersonId sorting, change from peek to foreach, added NPE check. Changed findPair algorythm. Code style fixes

lapitan committed:
Person: Fix

made fields final and removed set methods

Lada303 commented on ulabEdu/src/homework/ComplexExamples.java:
```diff
@@ -0,0 +1,257 @@
+      return false;
+    }
+    for (Person p : personsArray
+    ) {
```
вот это что? зачем это на новую строку

Lada303 commented on ulabEdu/src/homework/ComplexExamples.java:
```diff
@@ -0,0 +1,257 @@
+    if (personNullCheck(RAW_DATA)) {
+      List<Person> personsWithoutDuplicates = removeDuplicates(RAW_DATA);
+      List<Person> personsSortedById = sortPersonsById(personsWithoutDuplicates);
+      Map<String, Long> personsGroupedByName = groupPersonsByName(personsSortedById);
```
у тебя и было разбито на подметоды? Это излишне. все в одном стриме можно было сделать. Не надо усложнять реализацию )

Lada303 commented on ulabEdu/src/homework/ComplexExamples.java:
```diff
@@ -0,0 +1,257 @@
+  }
+
+  private static List<Person> removeDuplicates(Person[] personsArray) {
+    assert personsArray != null;
```
проверка на нулл делается в самом стриме - было показано на лекции. ты только все усложнил и запутал

lapitan committed:
week2Task1: Fix

remove functions for first task. Added nullCheck in stream for first task. Some codestyle fixes
